cardano-cli/src/Cardano/CLI/Types/Common.hs

```diff
@@ -122,6 +124,10 @@ newtype ProposalUrl = ProposalUrl
   { unProposalUrl :: L.Url
   } deriving (Eq, Show)
 
+newtype ProposalBinary = ProposalBinary
```

Comment:
That's funny, neither `ProposalText` or `ProposalBinary` are used in reading file: https://github.com/IntersectMBO/cardano-cli/blob/8def78a0ceca29027e7fbabcc9e5480b416eb231/cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/Hash.hs#L46

I guess the next step would be to make use of those newtypes. 😄

Reply:
They should be defined as type level tags.

Reply:
> They should be defined as type level tags.

Done in https://github.com/IntersectMBO/cardano-cli/pull/683